tests/e2e/test_historical_features.py

```diff
@@ -0,0 +1,124 @@
+import os
+import tempfile
+import uuid
+from datetime import datetime, timedelta
+from urllib.parse import urlparse
+
+import numpy as np
+import pandas as pd
+import pytest
+from google.protobuf.duration_pb2 import Duration
+from pandas._testing import assert_frame_equal
+
+from feast import Client, Entity, Feature, FeatureTable, FileSource, ValueType
+from feast.staging.storage_client import get_staging_client
+
+
+@pytest.fixture(scope="function")
+def staging_path(pytestconfig, tmp_path):
+    if pytestconfig.getoption("env") == "local":
+        return f"file://{tmp_path}"
+
+    staging_path = pytestconfig.getoption("staging_path")
+    return os.path.join(staging_path, str(uuid.uuid4()))
+
+
+def test_historical_features(feast_client: Client, staging_path: str):
+    customer_entity = Entity(
+        name="customer_id", description="Customer", value_type=ValueType.INT64
+    )
+    feast_client.apply_entity(customer_entity)
+
+    max_age = Duration()
+    max_age.FromSeconds(2 * 86400)
+
+    transactions_feature_table = FeatureTable(
+        name="transactions",
+        entities=["customer_id"],
+        features=[
+            Feature("daily_transactions", ValueType.DOUBLE),
+            Feature("total_transactions", ValueType.DOUBLE),
+        ],
+        batch_source=FileSource(
```

Comment:
I think we need to change the file source to have named arguments and not positional, even if they are required. The only positional argument should be the path, and it should come first. Otherwise its hard to understand what is being defined here.